test/common/stats/stat_test_utility.h

```diff
@@ -219,6 +219,29 @@ std::vector<uint8_t> serializeDeserializeNumber(uint64_t number);
 // Serializes a string into a MemBlock and then decodes it.
 void serializeDeserializeString(absl::string_view in);
 
+class TestSinkPredicates : public SinkPredicates {
+public:
+  ~TestSinkPredicates() override = default;
+  StatNameHashSet& sinkedStatNames() { return sinked_stat_names_; }
```

Comment:
WDYT of exposing this as an add(StatName stat_name) and has(Statname stat_name) rather then exposing the rep by returning the member variable reference?